backend/kernelCI_app/views/testsByTreeAndCommitHashView.py

```diff
@@ -11,73 +11,69 @@ def get(self, request, commit_hash: str | None):
         git_branch_param = request.GET.get("git_branch")
         boot_param_with_wildcard = "boot.%"
         boot_param_exact = "boot"
+
+        query = """
+            SELECT t.id, c.git_repository_url,
+                   c.git_commit_hash, t.id, t.build_id, t.start_time, t.duration,
+                   t.status as status, t.path, b.architecture, b.config_name,
+                   b.compiler, t.environment_misc,
+                   t.environment_comment, t.misc
+            FROM checkouts AS c
+            INNER JOIN builds AS b ON c.id = b.checkout_id
+            INNER JOIN tests AS t ON t.build_id = b.id
+            WHERE c.git_commit_hash = %s
+              AND c.origin = %s
+              AND c.git_repository_url = %s
+              AND c.git_repository_branch = %s
+        """
+
+        # Append the path filter conditionally
         if path_param == 'boot':
-            path_filter = "AND (t.path LIKE %s OR t.path = %s)"
-            params = ([commit_hash, origin_param, git_url_param,
-                       git_branch_param, boot_param_with_wildcard, boot_param_exact])
+            query += " AND (t.path LIKE %s OR t.path = %s)"
+            parameters = (
+                commit_hash, origin_param, git_url_param, git_branch_param,
+                boot_param_with_wildcard, boot_param_exact
+            )
         else:
-            path_filter = "AND (t.path NOT LIKE %s AND t.path != %s)"
-            params = ([commit_hash, origin_param, git_url_param,
-                       git_branch_param, boot_param_with_wildcard, boot_param_exact])
+            query += " AND (t.path NOT LIKE %s AND t.path != %s)"
+            parameters = (
```

Comment:
When I suggested my code I was thinking more in the sense of using a dictionary here than any other stuff there, but this works since it is not vulnerable to sql injection anymore